windows/utils/win_install_online_updates.yml

```diff
@@ -39,17 +51,77 @@
     - win_updates_install_result.found_update_count is defined
-    - win_updates_install_result.found_update_count | int != 0
     - win_updates_install_result.installed_update_count is defined
-    - win_updates_install_result.found_update_count | int == win_updates_install_result.installed_update_count | int
+  block:
+    # No updates found, so no updates installed
+    - name: "Set fact of Windows online updates install succeeds"
+      ansible.builtin.set_fact:
+        win_online_updates_succeed: true
+      when: win_updates_install_result.found_update_count | int == 0
+
+    # Updates found, updates installed
+    - name: "Installed update count is equal to found update count"
+      ansible.builtin.set_fact:
+        win_online_updates_succeed: true
+        win_online_updates_installed: true
+      when:
+        - win_updates_install_result.found_update_count | int != 0
+        - win_updates_install_result.installed_update_count | int == win_updates_install_result.found_update_count | int
+
+    # Updates found, while some not installed
+    - name: "Installed update count is not equal to found update count"
+      when:
+        - win_updates_install_result.found_update_count | int != 0
+        - win_updates_install_result.found_update_count | int > win_updates_install_result.installed_update_count | int
+      block:
+        - name: "Set fact of Windows online updates install status"
+          ansible.builtin.set_fact:
+            win_online_updates_installed: true
+          when: win_updates_install_result.installed_update_count | int != 0
+
+        - name: "Set fact of not installed updates list"
+          ansible.builtin.set_fact:
+            win_updates_not_installed: "{{ win_updates_install_result.updates.values() | selectattr('installed', 'equalto', false) }}"
+        - name: "Display the not installed updates list"
+          ansible.builtin.debug: var=win_updates_not_installed
+
+        - name: "Set fact of failure messages of not installed updates"
+          ansible.builtin.set_fact:
+            win_updates_failure_msgs: "{{ win_updates_not_installed | map(attribute='failure_msg') }}"
+        - name: "Set fact of the lengths of failure messages and not installed updates lists"
+          ansible.builtin.set_fact:
+            win_updates_failure_msgs_len: "{{ win_updates_failure_msgs | length }}"
+            win_updates_not_installed_len: "{{ win_updates_not_installed | length }}"
+
+        - name: "No failure messages for not installed updates"
+          ansible.builtin.debug:
+            msg: >-
+              {{ win_updates_not_installed_len - win_updates_failure_msgs_len }} updates
+              are not installed and without failure messages, which can not be ignored.
+          when: win_updates_failure_msgs_len != win_updates_not_installed_len
+
+        - name: "Get failure messages for all not installed updates"
+          when: win_updates_failure_msgs_len == win_updates_not_installed_len
```

Comment:
Will it be win_updates_failure_msgs_len !=0? I.e. as long as there is failure message, should check if it can be ignored.

Reply:
Ignore above comment.